tests/Microsoft.DotNet.Framework.Docker.Tests/ImageTests.cs

```diff
@@ -295,17 +296,24 @@ private static string GetRegistryName(string repo, string tag)
 
                 if (repoInfo["images"] != null)
                 {
-                    imageExistsInStaging = repoInfo["images"]
+                    isUsingCustomRegistry = repoInfo["images"]
                         .Cast<JProperty>()
                         .Any(imageInfo => imageInfo.Value["simpleTags"].Any(imageTag => imageTag.ToString() == tag));
                 }
                 else
                 {
-                    imageExistsInStaging = false;
+                    isUsingCustomRegistry = false;
                 }
             }
 
-            return imageExistsInStaging ? $"{Registry}/{RepoPrefix}" : "mcr.microsoft.com/";
+            if (isUsingCustomRegistry)
+            {
+                Debug.Assert(!String.IsNullOrEmpty(Registry), "Registry is not set.");
+                Debug.Assert(!String.IsNullOrEmpty(RepoPrefix), "RepoPrefix is not set.");
+                return $"{Registry}/{RepoPrefix}";
+            }
+            
+            return "mcr.microsoft.com/";
```

Comment:
I think the hard coded MCR registry reference should be changed to GetManifestRegistry().  e.g. this should only be specified in one place within the source.